tools/ExtensionsMetadataGenerator/src/ExtensionsMetadataGenerator.Console/ExtensionsMetadataGenerator.cs

```diff
@@ -183,5 +183,35 @@ public static string GetReflectionFullName(TypeReference typeRef)
         {
             return typeRef.FullName.Replace("/", "+");
         }
+
+        public class FunctionsAssemblyResolver : DefaultAssemblyResolver
+        {
+            private static HashSet<string> _trustedPlatformAssemblies = GetTrustedPlatformAssemblies();
+
+            private static HashSet<string> GetTrustedPlatformAssemblies()
+            {
+                var tpas = new HashSet<string>();
+                var data = AppContext.GetData("TRUSTED_PLATFORM_ASSEMBLIES");
+                foreach (var tpa in data.ToString().Split(";"))
+                {
+                    tpas.Add(tpa);
```

Comment:
Should we switch to the HashSet constructor which takes capacity value? That will prevent the auto-grow behavior(copy & allocation). I ran a mini benchmark (.NET 6 though): https://gist.github.com/kshyju/026fbce09291bbe2fe55d12b45c6b00e

Reply:
Oh, this project is .NET6.